sequence_task_1/student/include/impl.h

```diff
@@ -8,33 +8,85 @@ class StaticPlaylist
 {
 public:
     /** @todo Member traits */
+    typedef Song_t value_type;
+    typedef Song_t& reference;
+    typedef const Song_t& const_reference;
+    typedef typename Container::iterator iterator;
+    typedef typename Container::const_iterator const_iterator;
+    typedef ptrdiff_t difference_type;
+    typedef size_t size_type;
 
     /** @todo Iterators */
+    const_iterator begin() const
+    {
+        return m_tracklist.cbegin();
+    }
 
+    const_iterator end() const
+    {
+        return m_tracklist.cend();
+    }
     StaticPlaylist() = default;
 
     /** @todo Constructor from any reversible sequence container */
+    template<class GenericContainer>
+    StaticPlaylist(const GenericContainer& container)
+    {
+        m_tracklist.assign(container.rbegin(), container.rend());
+    }
 
     /** @todo Assignment from any reversible sequence container */
+    template<class GenericContainer>
+    StaticPlaylist<Container, Song_t>& operator=(const GenericContainer& other)
+    {
+        Container track_list;
+        track_list.assign(other.rbegin(), other.rend());
+
+        m_tracklist.swap(track_list);
```

Comment:
copy & swap is good, but usually you don't need in user code, you may utilize the assign